File: java/LeetCode0167.java
import java.util.Arrays;

public class LeetCode0167 {
    public int[] twoSum(int[] numbers, int target) {
        for (int i = 0;i < numbers.length - 1;++i) {
            int index = Arrays.binarySearch(numbers, i + 1, numbers.length, target - numbers[i]);
            if (index > i) {
                return new int[]{i + 1,index + 1};
            }
        }
        return new int[]{};
    }
}

File: cpp/L100/LeetCode0167.cc
#include <vector>
using std::vector;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        for (int low{0}, high(numbers.size() - 1);low < high;) {
            int x = numbers[low] + numbers[high];
            if (x == target) {
                return {low + 1, high + 1};
            } else if (x < target) {
                ++low;
            } else {
                --high;
            }
        }
        return {};
    }
};